impeller/renderer/backend/vulkan/formats_vk.h

```diff
@@ -466,7 +467,9 @@ constexpr vk::AttachmentDescription CreateAttachmentDescription(
   switch (kind) {
     case AttachmentKind::kColor:
       vk_attachment.initialLayout = current_layout;
-      vk_attachment.finalLayout = vk::ImageLayout::eColorAttachmentOptimal;
+      vk_attachment.finalLayout =
+          resolve_texture ? vk::ImageLayout::eGeneral
```

Comment:
The nitpick here is using "resolve_texture" as the reason to do this.

The not nitpick here is that I think we should go with eGeneral unless we know for sure that this is for the framebuffer. eGeneral should always be safe if I'm understanding the spec, whereas the eColorAttachmentOptimal is only safe if we know we're using this for framebuffer.